src/globals.d

```diff
@@ -14,26 +14,50 @@ import ddmd.root.array;
 import ddmd.root.filename;
 import ddmd.root.outbuffer;
 
-template xversion(string s)
-{
-    enum xversion = mixin(`{ version (` ~ s ~ `) return true; else return false; }`)();
-}
-
 private string stripRight(string s)
 {
     while (s.length && (s[$ - 1] == ' ' || s[$ - 1] == '\n' || s[$ - 1] == '\r'))
         s = s[0 .. $ - 1];
     return s;
 }
 
-enum IN_GCC     = xversion!`IN_GCC`;
+immutable bool TARGET_LINUX, TARGET_OSX, TARGET_FREEBSD, TARGET_OPENBSD,
+               TARGET_SOLARIS, TARGET_WINDOS, TARGET_POSIX;
 
-enum TARGET_LINUX   = xversion!`linux`;
-enum TARGET_OSX     = xversion!`OSX`;
-enum TARGET_FREEBSD = xversion!`FreeBSD`;
-enum TARGET_OPENBSD = xversion!`OpenBSD`;
-enum TARGET_SOLARIS = xversion!`Solaris`;
-enum TARGET_WINDOS  = xversion!`Windows`;
+shared static this()
```

Comment:
>  That is not answered anywhere in your three replies.

It is! In the first step.

> The first step is to move all uses of TARGET_XXX into the ddmd.target module.

Once that is done then as far as I'm concerned it is *gone* from the frontend.  Because `ddmd.target` is not a shared/common module.